feat: electron for snashot ui

ChaituVR opened the pull request:
### Summary

Closes: https://github.com/snapshot-labs/workflow/issues/87

### How to test

1. Details soon

### To-Do

- [ ] Test publishing to github
- [ ] Test auto-update

Sekhmet commented:
`electron:start` worked for me.

`electron:build` works as well, I tried installing and running app on macOS and it works, but I have few issues:
1. Logo we have doesn't fit with other app logos.
2. It took quite a long time for app to open (there was no window for 2 minutes it was just jumping on the taskbar).
3. Once it opened it's slow/unresponsive, now it just ended up as stuck for few minutes, if I click on some link it takes 30 seconds to navigate.

https://github.com/user-attachments/assets/9bb8ba6b-3aeb-4072-ab16-272ff20545e8

ChaituVR commented:
- Fixed the issue with huge logo
- @bonustrack  For macs using silicon-based chips, we should use `arm64` , not sure what is slowing down `x64` one, will try to install it on a mac using intel based chip
- Modified `yarn electron:start` to use localhost:8080 to support hot reload
- If we hide toolbar it will look like this, not sure if we want to go with this (we can also show them on hover if required)
<img width="213" height="183" alt="Untitled 7" src="https://github.com/user-attachments/assets/f4c7149c-81d9-4307-8c1f-3ade26137b5f" />

ChaituVR commented:
Moving to electron-forge for faster build time

ChaituVR commented:
We now use `electron-forge`, 
- Faster builds and smaller packages after few config changes
- Icon sizes for all platforms
- Tested on mac and windows but not tested on linux

Sekhmet commented on apps/ui/forge.config.js:
```diff
@@ -0,0 +1,40 @@
+export default {
+  packagerConfig: {
+    appBundleId: 'com.snapshot.app',
```
```suggestion
    appBundleId: 'box.snapshot.app',
```

I think we should use domain we own so either box.snapshot.app or org.snapshot.app

ChaituVR commented:
Fixed the issue with icon on desktop

I have created a separate issue for the new toolbar and scrollbar (because it may have many changes) https://github.com/snapshot-labs/workflow/issues/654

toolbar will look like this: 
<img width="1314" height="834" alt="Untitled 15" src="https://github.com/user-attachments/assets/868ad17b-58af-47af-872b-50f64dd3015e" />

We can merge this PR if the builds are working as expected

wa0x6e commented:
Maybe we should add `concurrently` to the dev dependency

Sekhmet reviewed (changes requested):
There was recent electron issue on macOS 26 that resulted in very bad performance: https://avarayr.github.io/shamelectron/

Can we update our electron version (we install `electron@38.1.0` which is affected). It was fixed in [v38.2.0](https://github.com/electron/electron/releases/tag/v38.2.0)

ChaituVR commented:
@Sekhmet Updated to 38.2 👍 also should we remove `^` for all npm libraries? or use `~` to allow only patch updates?

> I'm thinking, should we move logo a bit down (only on Electron in macOS only I guess) so it looks more evenly spaced out?

We have added a new titlebar for electron, which is handled in different issue 
https://github.com/snapshot-labs/workflow/issues/654
https://github.com/snapshot-labs/sx-monorepo/pull/1609

bonustrack commented:
I got some bug and suggestion

- When I open a proposal, click on "Votes" then click on "Overview" it got stuck, then if when I click on the space logo it show a broken layout like this:
<img width="1345" height="794" alt="image" src="https://github.com/user-attachments/assets/0e712083-4acb-4cdb-9c46-edc1a46446f3" />

- External link should open the default browser and not an Electron window.
- Can we define a min window size of 500 height and 400 width? Currently its possible to resize to this: 
<img width="131" height="88" alt="image" src="https://github.com/user-attachments/assets/06fe2bc4-4753-4325-b97c-bd121f3918e1" />

ChaituVR commented:
> When I open a proposal, click on "Votes" then click on "Overview" it got stuck, then if when I click on the space logo it show a broken layout

This happen on master branch as well, created a separate issue https://github.com/snapshot-labs/sx-monorepo/issues/1677

Fixed the other two suggestions

Sekhmet commented:
> I dont have the toolbar anymore using dev script

@bonustrack can you check if you don't have other dev server running when you run `dev:electron`? If you do electron specific dev-server will start at 8081 port, but electron will still use 8080 port.

bonustrack commented:
> > I dont have the toolbar anymore using dev script
> 
> @bonustrack can you check if you don't have other dev server running when you run `dev:electron`? If you do electron specific dev-server will start at 8081 port, but electron will still use 8080 port.

Good catch, it works now!

bonustrack commented:
The build on Netlify fail for some reason, even after rety

```
12:57:16 AM: Netlify Build                                                 
12:57:16 AM: ────────────────────────────────────────────────────────────────
12:57:16 AM: ​
12:57:16 AM: ❯ Version
12:57:16 AM:   @netlify/build 35.2.7
12:57:16 AM: ​
12:57:16 AM: ❯ Flags
12:57:16 AM:   accountId: 68cae686a18f563d780435aa
12:57:16 AM:   baseRelDir: true
12:57:16 AM:   buildId: 68fbbd6bd22b69466e160320
12:57:16 AM:   deployId: 68fbbd6bd22b69466e160322
12:57:16 AM:   packagePath: apps/ui
12:57:16 AM: ​
12:57:16 AM: ❯ Current directory
12:57:16 AM:   /opt/build/repo
12:57:16 AM: ​
12:57:16 AM: ❯ Config file
12:57:16 AM:   No config file was defined: using default values.
12:57:16 AM: ​
12:57:16 AM: ❯ Context
12:57:16 AM:   deploy-preview
12:57:16 AM: ​
12:57:16 AM: Build command from Netlify app                                
12:57:16 AM: ────────────────────────────────────────────────────────────────
12:57:16 AM: ​
12:57:16 AM: $ VITE_COMMIT_SHA=$COMMIT_REF turbo run build --filter=ui
12:57:17 AM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
12:57:17 AM: Attention:
12:57:17 AM: Turborepo now collects completely anonymous telemetry regarding usage.
12:57:17 AM: This information is used to shape the Turborepo roadmap and prioritize features.
12:57:17 AM: You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://turbo.build/repo/docs/telemetry
12:57:17 AM:   × No package found with name 'ui' in workspace
12:57:17 AM: ​
12:57:17 AM: "build.command" failed                                        
12:57:17 AM: ────────────────────────────────────────────────────────────────
12:57:17 AM: ​
12:57:17 AM:   Error message
12:57:17 AM:   Command failed with exit code 1: VITE_COMMIT_SHA=$COMMIT_REF turbo run build --filter=ui (https://ntl.fyi/exit-code-1)
12:57:17 AM: ​
12:57:17 AM:   Error location
12:57:17 AM:   In Build command from Netlify app:
12:57:17 AM:   VITE_COMMIT_SHA=$COMMIT_REF turbo run build --filter=ui
12:57:17 AM: ​
12:57:17 AM:   Resolved config
12:57:17 AM:   build:
12:57:17 AM:     command: VITE_COMMIT_SHA=$COMMIT_REF turbo run build --filter=ui
12:57:17 AM:     commandOrigin: ui
12:57:17 AM:     environment:
12:57:17 AM:       - NODE_OPTIONS
12:57:17 AM:       - REVIEW_ID
12:57:17 AM:       - VITE_ENABLED_NETWORKS
12:57:17 AM:       - VITE_HOST
12:57:17 AM:     publish: /opt/build/repo/apps/ui/dist
12:57:17 AM:     publishOrigin: ui
12:57:17 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
12:57:17 AM: Failing build: Failed to build site
12:57:18 AM: Finished processing build request in 2m24.943s
```
